### DISCUSSION TITLE
Issues about select all the link before el

### DISCUSSION THREAD

**Discussion by Ndmanh123** [clientIO/joint#1672] [Q&A] [answered]
Hi i working on a small dataflow and I want when i chose a element all the link before it should be change to another color link this and I also want the link to be  custom auto dodge the el 
![image](https://user-images.githubusercontent.com/104048582/169635908-a3266f74-a9dc-4cda-8f69-1549bf5ee6cd.png)

![image](https://user-images.githubusercontent.com/104048582/169635849-2bdfec14-77d7-4d16-b2c6-3e11e4d399cc.png)
cant i do it with jointjs and how cant i do it

**Comment by kumilingus** [comment, marked as answer]
1. To get all preceding links of an element run this:

```js
graph.getSubgraph([el, ...graph.getPredecessors(el)]).filter(cell => cell.isLink());
```

2. To make links automatically avoid elements use [manhattan router](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#routers.manhattan).



Here's a [CodePen](https://codepen.io/jointjs/pen/ZEryPjd/0a0972b6f3e399f3729d82f0da2d64b8) example.
